src/metabase/notification/seed.clj

```diff
@@ -1,124 +1,178 @@
+           :handlers      [{:active       true
+                            :channel_type :channel/email
+                            :channel_id   nil
+                            :template     {:name         "Slack Token Error Email template"
+                                           :channel_type "channel/email"
+                                           :details      {:type "email/handlebars-resource"
+                                                          :subject "Your Slack connection stopped working"
+                                                          :path "metabase/email/slack_token_error.hbs"
+                                                          :recipient-type "cc"}}
+                            :recipients   [{:type    :notification-recipient/template
+                                            :details {:pattern "{{context.admin_email}}" :is_optional true}}
+                                           {:type                 :notification-recipient/group
+                                            :permissions_group_id (:id (perms-group/admin))}]}]}]))
+
+
+(defn- cleanup-notification!
+  [internal-id existing-row]
+  (t2/delete! :model/Notification :internal_id internal-id)
+  (when-let [template-ids (->> existing-row :handlers (map (comp :id :template)) (filter some?) not-empty)]
+    (t2/delete! :model/ChannelTemplate :id [:in template-ids])))
 
 (defn- create-notification!
-  [{:keys [notification subscriptions handlers]}]
-  (let [handlers (for [handler handlers]
+  [notification]
+  (let [handlers (for [handler (:handlers notification)]
                    (if-let [template (:template handler)]
                      (-> handler
                          (dissoc :template)
                          (assoc :template_id (t2/insert-returning-pk!
                                               :model/ChannelTemplate
                                               template)))
                      handler))]
-    (models.notification/create-notification! notification subscriptions handlers)))
+    (models.notification/create-notification!
+     (dissoc notification :handlers :subscriptions)
+     (:subscriptions notification)
+     handlers)))
 
-(defn- seed-notification!
-  "Seed default notifications"
-  []
-  (log/info "Seeding default notifications")
-  (doseq [row @default-notifications]
-    (create-notification! row))
-  (log/infof "Seeded %d notifications" (count @default-notifications)))
+(defn- replace-notification!
+  "Replace an existing notification with a new one"
+  [existing-row {:keys [internal_id] :as new-row}]
+  (cleanup-notification! internal_id existing-row)
+  (create-notification! new-row))
+
+(defn- action
+  [existing-row new-row]
+  (cond
+    (nil? existing-row)
+    :create
+    ;; use json to compare to avoid any difference like keyword vs string
+    (not= (json/encode (sanitize-notification existing-row)) (json/encode (sanitize-notification new-row)))
```

Comment:
This is a fairly expensive way to do it, but I guess it's not a hot loop. Maybe extract to a helper method.